inject/src/main/java/io/micronaut/context/logging/DefaultLogbackLoggingConfigurer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.context.logging;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.ConsoleAppender;
+import ch.qos.logback.core.FileAppender;
+import io.micronaut.context.env.Environment;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Logback default implementation of {@link LoggingConfigurer}.
+ *
+ * @author Denis Stepanov
+ * @since 3.3
+ */
+public class DefaultLogbackLoggingConfigurer implements LoggingConfigurer {
+
+    @Override
+    public void apply(Environment environment) {
```

Comment:
what happens if logbook is not on the path?

Reply:
I would assume that `SoftServiceLoader` is going to exclude the entry.

Reply:
probably need to test that